src/adrs/0129349218/README.md

```diff
@@ -0,0 +1,532 @@
+
+Or the error happened in a downstream system, and you want to wrap the error
+with more context to the error and have a better understanding of what happened.
+
+The `causes` field should be used to wrap errors. The `causes` field contains a
+list of errors that caused this error.
+
+#### Form/Params/Command/Input Validation
+
+Everything starts with sending a command to the system, therefore, some errors
+are about the validation of such inputs.
+
+Giving feedback to the caller about what exactly went wrong is important to
+achieve clarity and also better user experience when dealing with inputs without
+pushing too much complexity to the clients.
+
+Having a `subject` field that identifies the input that caused the error is
+crucial to achieve such clarity. The `subject` field could be a JSON pointer
+that points to the input that caused the error relative to the command that
+caused the error. That means that when wrapping errors, the `subject` field
+is not absolute, and it is not aware of the wrapping.
+
+For example, imagine the following command:
+
+```js
+const command = {amount: '1000', currency: 'CUP'};
+```
+
+Which will produce the following error:
+
+```js
+const error = {
+  subject: '/data',
+  domain: 'com.mybussines.myapp',
+  type: 'bad_request',
+  causes: [
+    {
+      subject: '/currency',
+      domain: 'com.mybussines.myapp',
+      type: 'invalid_currency',
+      data: {valid_currencies: ['USD']},
+      // ommited data ...
+    },
+  ],
+};
+```
+
+You could group by the `subject` and present the error to the user in a more
+friendly way in the UI such as displaying the error message next to the input
+that caused the error. Since the `subject` is relative to the command payload,
+you may need to merge the `subject` when wrapping errors.
+
+
+## Pending Concerns
+
+- Having infra-level `info` where you could use the `metadata` to identify the
+  context in which the error happened introduces a level of complexity that
+  it is desired to be avoided. The `domain` and `type` fields should be enough
+  to identify the context in which the error happened. Example:
+  `googleapis.com` domain and `NOT_FOUND` reason. Most likely the `metadata`
+  field will be used to provide more context to the error message, example:
+  `resource_type`. If `id` must be unique across the entire domain, then
+  it becomes a coordination problem, and **it is desired to avoid it**.
+- Message placeholder comes from C# ecosystem, what is not desired here is to
+  introduce security vulnerabilities, look at what happened to Java log4j
+  vulnerability. So, is this desired?
+- Should Message Placeholder allow to replace with the `info` `metadata`?
+  Probably not, since whoever defines the `info` and its structure owns the
+  data, and the structure of the error
+- `source?: Source;` I am not sure about the naming.
+- Under `info.data`
+
+## Resolution
+
+- You **MUST** use the `application/universal-error+json` meme when sending the
```

Comment:
What is this?

Reply:
Fixed, I meant to say `mime` (too much memes in my life that)